lib/content-services/src/lib/search/components/search-logical-filter/search-logical-filter.component.ts

```diff
@@ -0,0 +1,142 @@
+    encapsulation: ViewEncapsulation.None
+})
+export class SearchLogicalFilterComponent implements SearchWidget, OnInit {
+    private searchCondition: LogicalSearchCondition;
+    private reset$ = new Subject<void>();
+
+    id: string;
+    settings?: SearchWidgetSettings;
+    context?: SearchQueryBuilderService;
+    startValue: string;
+    displayValue$: Subject<string> = new Subject<string>();
+    resetObservable = this.reset$.asObservable();
+    LogicalSearchFields = LogicalSearchFields;
+
+    constructor(private translationService: TranslationService) {}
+
+    ngOnInit(): void {
+        this.searchCondition = { matchAll: [], matchAny: [], exclude: [] };
+        this.updateDisplayValue();
+    }
+
+    onPhraseChange(phrases: string[], field: LogicalSearchFields) {
+        this.searchCondition[field] = phrases;
+        this.updateDisplayValue();
+    }
+
+    submitValues() {
+        if (this.hasValidValue() && this.id && this.context && this.settings && this.settings.field) {
+            this.updateDisplayValue();
+            const fields = this.settings.field.split(',').map((field) => field += ':');
+            let query = '';
+            Object.keys(this.searchCondition).forEach((key) => {
+                if (this.searchCondition[key].length > 0) {
+                    let connector = '';
+                    let subQuery = '';
+                    switch(key) {
+                        case LogicalSearchFields.MATCH_ALL:
+                            connector = 'AND';
+                            break;
+                        case LogicalSearchFields.MATCH_ANY:
+                            connector = 'OR';
+                            break;
+                        case LogicalSearchFields.EXCLUDE:
+                            connector = 'AND NOT';
+                            break;
+                        default:
+                            break;
+                    }
+                    fields.forEach((field) => {
+                        subQuery += subQuery === '' ? '' : key === LogicalSearchFields.EXCLUDE ? ' AND ' : ' OR ';
+                        let fieldQuery = '(';
+                        this.searchCondition[key].forEach((phrase: string) => {
+                            const refinedPhrase = '\"' + phrase + '\"';
+                            fieldQuery += fieldQuery === '(' ?
+                                `${key === LogicalSearchFields.EXCLUDE ? 'NOT ' : ''}${field}${refinedPhrase}` :
+                                ` ${connector} ${field}${refinedPhrase}`;
+                        });
+                        subQuery += `${fieldQuery})`;
+                    });
+                    query += query === '' ? `(${subQuery})` : ` AND (${subQuery})`;
+                    subQuery = '';
+                }
+            });
+            this.context.queryFragments[this.id] = query;
+            this.context.update();
+        }
+    }
+
+    updateDisplayValue(): void {
```

Comment:
can probably be private, if not used outside

Reply:
Right, I made it private thanks